tests/integration/ceph_base_object_test.go

```diff
@@ -142,6 +92,74 @@ func runObjectE2ETest(helper *clients.TestClient, k8sh *utils.K8sHelper, s suite
 	assert.True(s.T(), k8sh.CheckPodCountAndState("rook-ceph-mgr", namespace, 1, "Running"))
 	logger.Infof("Ceph MGRs are running alright")
 
+	/* Testing creation/deletion of objects using Object Bucket Claim */
+	logger.Infof("Step 3 : Create Object Bucket Claim with reclaim policy delete")
+
+	cobErr := helper.BucketClient.CreateBucketStorageClass(namespace, storeName, storageClassName, "Delete", region)
+	require.Nil(s.T(), cobErr)
+	cobcErr := helper.BucketClient.CreateObc(obcName, storageClassName, bucketname, true)
+	require.Nil(s.T(), cobcErr)
+
+	logger.Infof("Checking to see if the obc,secret,cm has been created")
+	for i = 0; i < 4 && helper.BucketClient.ObcExists(obcName) == false; i++ {
+		logger.Infof("(%d) obc check sleeping for 5 seconds ...", i)
+		time.Sleep(5 * time.Second)
+	}
+	assert.True(s.T(), helper.BucketClient.ObcExists(obcName))
+
+	logger.Infof("Check whether bucket got created or not")
+	context := k8sh.MakeContext()
+	rgwcontext := rgw.NewContext(context, storeName, namespace)
+	bkt, _, _ := rgw.GetBucket(rgwcontext, bucketname)
+	require.Equal(s.T(), bkt.Name, bucketname)
+	logger.Infof("Done creating OBC successfully")
+
+	logger.Infof("Step 4 : Create s3 client")
+
+	s3endpoint, _ := helper.ObjectClient.GetEndPointUrl(namespace, storeName)
+	s3AccessKey, _ := helper.BucketClient.GetAccessKey(obcName)
+	s3SecretKey, _ := helper.BucketClient.GetSecretKey(obcName)
+	s3client := utils.CreateNewS3Helper(s3endpoint, s3AccessKey, s3SecretKey)
+	logger.Infof("endpoint (%s) Accesskey (%s) secret (%s)", s3endpoint, s3AccessKey, s3SecretKey)
+
+	logger.Infof("Step 5 : Put Object on bucket")
+
+	_, poErr := s3client.PutObjectInBucket(bucketname, objBody, objectKey, contentType)
+	require.Nil(s.T(), poErr)
+
+	logger.Infof("Step 6 : Get Object from bucket")
+
+	read, err := s3client.GetObjectInBucket(bucketname, objectKey)
+	require.Nil(s.T(), err)
+	require.Equal(s.T(), objBody, read)
+	logger.Infof("Object Created and Retrieved on bucket successfully")
+
+	logger.Infof("Step 7 : Delete Object on bucket")
+
+	_, delobjErr := s3client.DeleteObjectInBucket(bucketname, objectKey)
+	require.Nil(s.T(), delobjErr)
+	logger.Infof("Object deleted on bucket successfully")
+
+	logger.Infof("Delete Object Bucket Claim")
+	dobcErr := helper.BucketClient.DeleteObc(obcName, storageClassName, bucketname, true)
+	require.Nil(s.T(), dobcErr)
+	logger.Infof("Checking to see if the obc,secret,cm has been deleted")
+	for i = 0; i < 4 && helper.BucketClient.ObcExists(obcName) == true; i++ {
```

Comment:
Similarly to above, we need the retry for bucket deletion, not obc deletion?

Reply:
This is a bit different than OBC creation. The OBC is the owner of the Secret and CM, both of which have a finalizer on them. [Future note: the obc will also have a finalizer added once library issue #131 is fixed.] An OBC is marked for deletion but the delete is deferred until secret, CM, and OB are deleted, all of which occur _after_ the provisioner has a chance to delete the bucket. So it may be reasonable to have a loop here.

Reply:
`ObcExists` returns false if _any_ of the Secret, CM or OBC are _not_ present. True is returned only if all 3 are present, so this loop stops as soon as the 1st of the checked resources is deleted. Instead, the absence of all 3 needs to trigger the stop of the for loop and be handled in the assert below.